docs/source/blueprints/BP005.rst

```diff
@@ -0,0 +1,291 @@
+   also day to day tasks. Splitting the repository can increase the adoption by
+   other projects that will choose to use this standalone repo because of its
+   inherent value.
+
+2. Better reuse of libraries and code between Avocado, Avocado-VT and third
+   party projects
+
+   Currently we are duplicating efforts by maintaining common libraries in Avocado
+   and Avocado-VT. By concentrating those libraries in a standalone repository, we
+   will make sure they will be well suited for different use cases.  By having
+   more users, the quality and collaboration of those libraries will be improved.
+   Also, third-party software will be able to import those libraries without the
+   need of having the entire Avocado framework as a dependency. This will reduce
+   the learning curve of future potential users that don't need the entire
+   framework.
+
+3. Making it clear to our users that Avocado framework is not Python-centric:
+
+   Currently, those utility libraries can only be used by Python tests and the
+   Avocado framework was designed to support tests written in multiple
+   languages.  By having those libraries outside the core harness codebase we
+   reinforce the language agnostic characteristic of Avocado.
+
+4. Reduce traffic in the main framework repository and allow for a scalable
+   community on its own standalone project
+
+   As evidenced by “avocado-misc-tests”, a more focused standalone project will
+   attract a larger number of similar minded developers and users. By being a
+   simpler standalone repository, with its own specific contributions
+   guidelines, the entry barriers are also lowered.
+
+
+Specification
+*************
+
+To avoid the need to change our current imports, our first attempt was to keep
+the "avocado.utils" namespace regardless of the location of its codebase by
+using the Implicit Package Namespace (PEP-420) approach. Namespace packages are
+a mechanism for splitting a single Python package across multiple directories
+on disk. Unfortunately, to be able to convert the avocado.utils.* modules and
+packages into a namespace package, we would need to remove the `__init__.py`
+files from multiple places. That would break, among other things, the current
+Avocado egg based deployment model used on spawners such as podman.
+
+Because of that, we decided to have a new namespace for the utility libraries.
+We ran a poll to choose the name and `autils` (Avocado Utils) was the winner.
+So our goal is to host common libraries under `avocado-project/autils`
+repository on GitHub and make them also available on PyPI under `python-autils`
+package name, since the `autils` name is unavailable.
+
+Utility onboarding
+==================
+
+The primary goal of autils is to be a central place for modules that ease the
+interaction with system level features and interfaces.  As such, if a user
+needs to interact with such a system level feature or interface, he/she may
+propose a new module.
```

Comment:
That's a tough question to answer in such a generic way, because the PoV of one user may not be the same of the maintainer and vice-versa.

As a general rule, maintainers should indeed try to avoid having duplicate modules in autils if well established options are available elsewhere.  But, there may be situations in which adding a new module in autils:

* Will work better with other autils libraries (for instance, allowing remote execution transparently)
* Can simplify deployment (if autils' version has leaner requirements)
* Follow the other autils APIs, for a more consistent experience
* Will guarantee/advertise the testing of that module in different platforms

In short, I think using common sense and some differentiation checks (like the examples above) may help a maintainer to decide if a new module is worth having or not.  Of course, people proposing such a module would also need to take responsibility on its maintenance.

Reply:
Sounds perfect, thanks!

Reply:
Thanks @clebergnu for clarifying.

I'm resolving this conversation so we can approve this Blueprint. Please reopen a new thread if you would like to get something fixed on this draft.